### PR TITLE
Update losses implementation

### PR DESCRIPTION
**Pull request by alanyee**
-Various replacements of deprecated functions with equivalent tf.losses
-Minor HTTPS calls

### CODE REVIEW

File: tensorflow/contrib/learn/python/learn/estimators/logistic_regressor_test.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
 #
-#     http://www.apache.org/licenses/LICENSE-2.0
+#     https://www.apache.org/licenses/LICENSE-2.0
```
Comment (vrv) [inline tensorflow/contrib/learn/python/learn/estimators/logistic_regressor_test.py]: For better or worse, the original link is what is in the Apache license.  can we keep it the way it is to keep the headers identical?

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Comment by alanyee** [comment]
@vrv could you test it now?

**Comment by vrv** [comment]
re-testing.  @tensorflow-jenkins test this please

**Comment by vrv** [comment]
Can you remove the unrelated changes that aren't about losses?

**Comment by alanyee** [comment]
@vrv i have reverted unrelated changes. I will just submit a separate PR of them.

**Comment by vrv** [comment]
Thanks -- some of those changes seemed unnecessary so I didn't want to delay this PR for that.

**Pull request merged**